object/datastore.go

```diff
@@ -58,6 +58,8 @@ func (e DatastoreNoSuchFileError) Error() string {
 
 type Datastore struct {
 	Common
+
+	DatacenterPath string
```

Comment:
Started on that route initially, but opted to keep the diff a bit smaller and ideally there won't be other types that need this path.  Datastore is the only type currently that uses HTTP without the vmodl.